### DISCUSSION TITLE
Is there any way to keep the parameters hint?

### DISCUSSION THREAD

**Discussion by Tudyx** [helix-editor/helix#8159] [Q&A]
When I call a function with a variable as parameter, I often make sure that the variable as the same name as the parameter for consistency.
For instance, if i have this function:

```rust
fn foo(bar : i32) {
    // ..
}
```
I will name the variable `bar`:

```rust
let bar = 2;
foo(bar);
```

Now, if start typing the name of the function and use auto-completion, it will automatically fulfill the function with a parameters name bar, which is great. But when I press enter (or escape/ Ctrl-c) the parameter hint will be removed. That's annoying because i will have to rewrite the parameters name by hand. I understand that's the behavior we want most of the time.
Is there a key combination that I can hit to keep the suggested parameters? I couldn't find anything in the doc and opened discussions/issues.

**Comment by araruna** [comment]
It's not just in this situation. In the context of `rust-analyzer` most of the time it provides a lot of what I want as hints, like in the following asciinema cast, or other cases, like automatically suggesting borrowing the right variable in the right way. But it just vanishes...

https://asciinema.org/a/RGI1BtOzqikk4YUZw8DQlG60S

I +1  this question in the hopes someone has the solution.

**Reply by gabydd** [reply]
https://github.com/helix-editor/helix/pull/9801 will keep placeholders at the tab stop, and then you can choose if you want to overwrite them
